### PR TITLE
DiagnosticSettingsFilter fails to correctly apply the absent operator

### PR DESCRIPTION
**Pull request by h-dub**
In response to [#6129](https://github.com/cloud-custodian/cloud-custodian/issues/6129)
The DiagnosticSettingsFilter fails to apply the absent operator. When there are no diagnostic settings configured on a resource, the code is not sending in an empty item for absent to match against.

```
policies:
  - name: az-publicip-without-ddosnotify
    resource: azure.publicip
    filters:
      - or:
        - type: diagnostic-settings
          key: logs[?category == 'DDoSProtectionNotifications'][].enabled
          value: absent
        - type: diagnostic-settings
          key: logs[?category == 'DDoSProtectionNotifications'][].enabled
          value: False
          op: in
          value_type: swap
```

Terraform/CCPolicy repros can be found here:
https://github.com/h-dub/cloud-custodian/pull/3

### CODE REVIEW

**Comment by h-dub** [comment]
coverage fixed

File: tools/c7n_azure/c7n_azure/filters.py [hunk 1]
```diff
@@ -426,7 +426,9 @@ def process_resource_set(self, resources):
         for resource in resources:
             settings = client.diagnostic_settings.list(resource['id'])
             settings = [s.as_dict() for s in settings.value]
-
+            # put an empty item in when no diag settings, so the absent operator can function
+            if(not settings):
```
Comment (kapilt) [inline tools/c7n_azure/c7n_azure/filters.py:430]: nit - i'm somewhat surprised this passed flake8 linting, more idiomatic would just be

`if not settings:`

Comment (h-dub) [reply tools/c7n_azure/c7n_azure/filters.py]: fixed

**Review by kapilt — APPROVED** [review]
lgtm, thanks

**Pull request merged**